scripts/packages/VSCodeServer/src/verionslens.jl

```diff
@@ -0,0 +1,38 @@
+include("../../RegistryQuery/RegistryQuery.jl")
+
+
+JSONRPC.@dict_readable struct LensParams <: JSONRPC.Outbound
+    name::String
+    uuid::String
+end
+
+JSONRPC.@dict_readable struct LensResponse <: JSONRPC.Outbound
+    latest_version::Union{String,Nothing}
+    url::Union{String,Nothing}
+    registry::Union{String,Nothing}
+end
+LensResponse(m::RegistryQuery.PkgMetadata) = LensResponse(m.latest_version, m.url, m.registry)
+
+
+registries = RegistryQuery.reachable_registries()
+function lens_request(conn, params::LensParams)
+    for registry in registries
+        uuid = nothing
+        try
+            uuid = UUID(params.uuid)
+        catch
+            uuid = RegistryQuery.uuids_from_name(registry, params.name)[1]
+        end
+
+        metadata =  RegistryQuery.get_pkg_metadata(registry, uuid)
+        # If the pakcage is knwown, return, i.e., don't query other registries.
+        if metadata !== RegistryQuery.UnknownPkgMetadata
+            return LensResponse(metadata)
+        end
+    end
+
+    # If the package isn't found in all registries, return unknown package.
+    return LensResponse(RegistryQuery.UnknownPkgMetadata)
+end
+
+const lens_request_type = JSONRPC.RequestType("lens/pkgVersions", LensParams, LensResponse)
```

Comment:
So it seems to me that all of this code should definitely not live in the REPL process. The REPL might not be running when someone opens a `Project.toml`, but that shouldn't be a reason that this functionality doesn't work, right?

I think if incorporating this into the LS is too involved right now, then it should still live in its own process (which, though, doesn't seem less work than just moving this functionality into the LS).

Reply:
@davidanthoff, please take a look at https://github.com/julia-vscode/LanguageServer.jl/pull/1055.